plaso/parsers/windefenderhistory.py

```diff
@@ -0,0 +1,342 @@
+# -*- coding: utf-8 -*-
+"""Parser for Windows Defender DetectionHistory files."""
+
+import os
+
+from datetime import datetime
+from datetime import timedelta
+
+from dtfabric.runtime import data_maps as dtfabric_data_maps
+from dfdatetime import time_elements as dfdatetime_time_elements
+
+from plaso.containers import events
+from plaso.containers import time_events
+
+from plaso.lib import definitions
+from plaso.lib import dtfabric_helper
+from plaso.lib import specification
+
+from plaso.parsers import interface
+from plaso.parsers import manager
+
+
+class WinDefenderHistoryEventData(events.EventData):
+  """Windows Defender Detection History event data.
+
+  Attributes:
+    container_filename(str): If the detected file was in a container,
+      its location.
+    extra(str): Comma separated list of extra detection locations.
+    filename(str): Filename that the threat was detected in.
+    host_and_user(str): Endpoint name in DOMAIN\\USER format.
+    process(str): The process that cased the detection.
+    sha256(str): Hash of the file.
+    threat_name(str): The threat that was detected.
+    web_filename(str): If the detected file was downloaded from the web,
+        its URI.
+  """
+
+  DATA_TYPE = 'av:defender:detection_history'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(WinDefenderHistoryEventData, self).__init__(
+        data_type=self.DATA_TYPE)
+    self.container_filename = None
+    self.extra = None
+    self.filename = None
+    self.host_and_user = None
+    self.process = None
+    self.sha256 = None
+    self.threat_name = None
+    self.web_filename = None
+
+
+class WinDefenderHistoryParser(
+    interface.FileObjectParser, dtfabric_helper.DtFabricHelper):
+  """Parses the Windows Defender History Log."""
+
+  NAME = 'windefenderhistory'
+
+  _FILE_SIGNATURE = 'Magic.Version:1.2'
+
+  _VALUE_DESCRIPTIONS = [
+      {0: "Threat identifier", 1: "Identifier"},
+      {0: "UnknownMagic1", 1: "Threat name", 4: "Category"},
+      {0: "UnknownMagic2",
+       1: "Resource type",
+       2: "Resource location",
+       4: "Threat tracking data size",
+       5: "Threat tracking data",
+       6: "Last threat status change time",
+       12: "Domain user1",
+       14: "Process name",
+       18: "Initial detection time",
+       20: "Remediation time",
+       24: "Domain user2"}]
+
+  _DEFINITION_FILE = os.path.join(
+      os.path.dirname(__file__), 'detection_history.yaml')
+
+  @classmethod
+  def GetFormatSpecification(cls):
+    """Retrieves the format specification.
+
+    Returns:
+      FormatSpecification: format specification.
+    """
+    format_specification = specification.FormatSpecification(cls.NAME)
+    format_specification.AddNewSignature(
+      b'M\x00a\x00g\x00i\x00c\x00.\x00V\x00e\x00r\x00s\x00i\x00o\x00n\x00:'
```

Comment:
use 4 space continuation indentations

this can be also written as `self._FILE_SIGNATURE.encode('utf-16-le')` which is more readable